sim/dex.ts

```diff
@@ -132,6 +134,63 @@ const Natures: {[k: string]: Nature} = {
 };
 export const toID = Data.toID;
 
+// interface for the builder.
+interface FormatSection {
+	section: string;
+	column?: number;
+	formats: FormatData[];
+}
+
+// function for merging the two lists
+function merge(
+			   main: FormatList[],
+			   side: FormatList[]
+): FormatList[] {
+	// result that is return and makes the actual list for formats.
+	let result: FormatList[] = [];
+
+	// used as a intermediary to build the final list.
+	const build: FormatSection[] = [];
+
+	// used to track location to keep formats under their sections.
+	let loc = -1;
+
+	// populates the original sections and formats easily
+	// there should be no repeat sections at this point.
+	for (const element of main) {
+		if (element.section) {
+			build.push({section: element.section, column: element.column, formats: []});
+			loc++;
+		} else if ((element as FormatData).name) {
+			build[loc].formats.push((element as FormatData));
+		}
+	}
+
+	// merges the second list the hard way. Accounts for repeats.
+	for (const element of side) {
+		// finds the section and makes it if it doesn't exist.
+		if (element.section) {
+			loc = build.findIndex(x => x === element.section);
+
+			// if it's new it makes a new entry.
+			if (loc === -1) {
+				build.push({section: element.section, column: element.column, formats: []});
+			}
+		} else if ((element as FormatData).name) { // otherwise, adds the element to its section.
+			build[loc].formats.push(element as FormatData);
+		}
+	}
+
+	// builds the final result.
+	for (const element of build) {
+		// adds the section to the list.
+		result.push({section: element.section, column: element.column});
+		result = result.concat(element.formats);
```

Comment:
```suggestion
		result.push({section: element.section, column: element.column}, ...element.formats);
```

Reply:
I forgot about the ... thing before lists, thank you for reminding me.